android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -272,33 +272,65 @@ private String getLackPermissionToDownloadMessage() {
   }
 
   @ReactProp(name = "javaScriptEnabled")
-  public void setJavaScriptEnabled(WebView view, boolean enabled) {
-    view.getSettings().setJavaScriptEnabled(enabled);
+  public void setJavaScriptEnabled(RNCWebView view, boolean enabled) {
+    view.ifHasInternalWebView(webView -> webView.getSettings().setJavaScriptEnabled(enabled));
   }
 
   @ReactProp(name = "setBuiltInZoomControls")
-  public void setBuiltInZoomControls(WebView view, boolean enabled) {
-    view.getSettings().setBuiltInZoomControls(enabled);
+  public void setBuiltInZoomControls(RNCWebView view, boolean enabled) {
+    view.ifHasInternalWebView(webView -> webView.getSettings().setBuiltInZoomControls(enabled));
   }
 
   @ReactProp(name = "setDisplayZoomControls")
-  public void setDisplayZoomControls(WebView view, boolean enabled) {
-    view.getSettings().setDisplayZoomControls(enabled);
+  public void setDisplayZoomControls(RNCWebView view, boolean enabled) {
+    view.ifHasInternalWebView(webView -> webView.getSettings().setDisplayZoomControls(enabled));
   }
 
   @ReactProp(name = "setSupportMultipleWindows")
-  public void setSupportMultipleWindows(WebView view, boolean enabled){
-    view.getSettings().setSupportMultipleWindows(enabled);
+  public void setSupportMultipleWindows(RNCWebView view, boolean enabled){
+    view.ifHasInternalWebView(webView -> webView.getSettings().setSupportMultipleWindows(enabled));
+  }
+
+  @ReactProp(name = "webViewKey")
+  public void setWebViewKey(RNCWebView view, String webViewKey) {
+    Map<String, WebView> internalWebViewMap = RNCWebViewMapManager.INSTANCE.getInternalWebViewMap();
+    Map<String, RNCWebView> rncWebViewMap = RNCWebViewMapManager.INSTANCE.getRncWebViewMap();
+
+    // If there is an existing RNCWebView that has an internal webview, re-attach it to this view
+    if (rncWebViewMap.containsKey(webViewKey)) {
+      RNCWebView existingView = rncWebViewMap.get(webViewKey);
+      InternalWebView existingWebView = existingView.detachWebView();
+      view.attachWebView(existingWebView);
+
+      // The chrome client was originally setup on instance creation but might be pointing to the wrong webview
+      // so it's reset here.
+      // Not entirely sure why there is a single instance of the webchrome client for all webviews?
```

Comment:
I don't think there is. I assume you're referring to `RNCWebViewManager` keeping its own copy of `mWebChromeClient`, but as far as I can tell the only thing that's used for is to call `onHideCustomView` on the "previous" instance when creating a new instance...? But it looks to me like a new instance is created (sometimes multiple times) for each WebView...?

I'm not familiar with the `onHideCustomView` logic.... I'm _guessing_ that implementation just never considered the possibility of having more than one WebView...